packages/react-dev-overlay/src/internal/container/BuildError.tsx

```diff
@@ -61,4 +67,97 @@ export const styles = css`
   .nextjs-container-build-error-body small {
     color: #757575;
   }
+
+  .nextjs-container-build-error-version-status {
+    font-weight: bold;
+  }
+  .nextjs-container-build-error-version-status span {
+    display: inline-block;
+    width: 10px;
+    height: 10px;
+    border-radius: 5px;
+    background: #eaeaea;
+    margin-right: 6px;
+  }
+
+  .nextjs-container-build-error-version-status span.fresh {
+    background: #50e3c2;
+  }
+
+  .nextjs-container-build-error-version-status span.outdated {
+    background: #e00;
+  }
+
+  .nextjs-container-build-error-version-status span.stale {
+    background: #f5a623;
+  }
 `
+declare global {
+  interface Window {
+    next: { version: string; latest: string }
+  }
+}
+
+type Staleness = 'fresh' | 'stale' | 'outdated'
+
+const NextStatus = () => {
+  const [staleness, setStaleness] = React.useState<Staleness>()
+  React.useEffect(() => {
+    ;(async () => {
+      try {
+        const installedVersion = window.next.version.replace(/-canary.*/, '')
+
+        const latestVersion = await getLatest()
+        console.log(latestVersion, installedVersion)
+
+        const result = compareVersions(installedVersion, latestVersion)
+        setStaleness(result)
+      } catch {}
+    })()
+  }, [])
+
+  const title = {
+    fresh: 'Next.js is up to date!',
+    stale: 'Next.js is out of date, update recommended!',
+    outdated: 'Next.js is out of date, update necessary!',
+    undetermined: 'Could not determine Next.js version status.',
+  }[staleness ?? 'undetermined']
+
+  return (
+    <div title={title} className="nextjs-container-build-error-version-status">
+      <span className={staleness} />
+      Next.js {window.next.version}
+    </div>
+  )
+}
+
+async function getLatest() {
+  return '12.1.6'
+  if (window.next.latest) {
+    return window.next.latest
+  }
+
+  const res = await fetch(
+    'https://api.github.com/repos/vercel/next.js/releases/latest'
+  )
```

Comment:
What about npm registry API?

https://registry.npmjs.org/next/latest

It will make sure to use the actual published version (maybe due to a CI bug or npm outage, it might be possible that only the GitHub release is created but no package is published on npm).

Reply:
Thanks, I was thinking about that as well, but it does not seem to give us the time of the publish. We would like to show information on how much time has passed between the installed and latest version to give a better indication of the urgency of upgrading.

I just got this from https://github.com/vercel/next.js/blob/canary/packages/next/cli/next-info.ts anyway, will iterate on it, thanks!